File: DevNews/DevNews.Server/Article.Web.Server/model/channel.ts
export type ChannelModel = {
    link: string;
    title: string;
    avatar: string;
    name: string;
    token: string;
    id: string;
}

export type SaveFile = {
    base64: string;
    path: string;
}

export type ChannelAdmin = {
    userName: string;
    email: string;
    avatar: string;
    token: string;
}

export type ChannelInfo = {
    channel: ChannelModel;
    owner: ChannelAdmin;
    isAdmin: boolean;
    mute: boolean;
    isIn: boolean;
}

export type FileModel = {
    base64: string;
    ogName: string;
    type: string;
}

export type SaveChannelFile = {
    file: FileModel;
    objId: string;
}

File: DevNews/ViewModel/Channel/Channel.cs
using ViewModel.File;

namespace ViewModel.Channel;

public record ChannelViewModel(string Token, string Link, string Name, IEnumerable<Entity.Article.TFile> Avatar);

public record ChannelPreviewViewModel(string Token, string Name, IEnumerable<Entity.Article.TFile> Avatar, int NewPosts);

public record GetChannelsResponse(ChannelsStatus Status, IEnumerable<ChannelPreviewViewModel> Channels);

public record UpsertChannelViewModel(string Name, string Link, IEnumerable<Guid> Categories, UploadFileViewModel Avatar);

public record UpsertChannelResponse(ChannelsStatus Status, ChannelPreviewViewModel Channel);

public record ChannelInfoViewModel(ChannelsStatus Status, ChannelViewModel Channel, OwnerViewModel Owner, bool IsAdmin, bool IsIn);

public record OwnerViewModel(string Email, string Token, string UserName, IEnumerable<Entity.Article.TFile> Avatar);

public enum ChannelsStatus
{
    Success = 0,
    UserNotFound = 1,
    Exception = 2,
    ChannelNotFound = 3
}